core/src/main/java/org/infinispan/eviction/impl/PassivationManagerImpl.java

```diff
@@ -55,6 +59,11 @@
    public void start() {
       enabled = !persistenceManager.isReadOnly() && cfg.persistence().passivation() && cfg.persistence().usingStores();
       if (enabled) {
+         if (!(persistenceManager instanceof PassivationPersistenceManager)) {
+            throw new CacheException("Passivation is enabled, but PersistenceManager is not instanceof " +
+                  "PersistancePassivationManager was " + persistenceManager.getClass());
+         }
+         passivationPersistenceManager = (PassivationPersistenceManager) persistenceManager;
```

Comment:
If passivation is enabled, I thought the code and exception explained it well enough. Did you want me to add a comment?

Reply:
Doesn't the logic in `EmptyConstructorNamedCacheFactory` around wrapping a `PersistenceManager` prevent this state from every being reached?

Because you have:
```java
         if (configuration.persistence().usingStores()) {
            PersistenceManagerImpl realPersistenceManager = new PersistenceManagerImpl();
            if (configuration.persistence().passivation()) {
               return new PassivationPersistenceManager(realPersistenceManager);
            }
            return realPersistenceManager;
         }
```

Reply:
That is the goal :) It is more to make sure there wasn't a bug, I could remove the instanceof check and just have it throw a ClassCastException instead? The other issue was tests can inject a different PersistenceManager implementation that isn't this.

Reply:
e.g. https://github.com/infinispan/infinispan/pull/7168/files#diff-e16dea2326f5b31bf9d28dff9a9bb5b4R97

Reply:
Did you mean the exception can happen or the fact that it is a `PassivationPersistenceManager`?

Reply:
> The other issue was tests can inject a different PersistenceManager implementation that isn't this.

Fair enough, I didn't really consider tests and I guess a more helpful exception can't hurt.

Reply:
> Did you mean the exception can happen or the fact that it is a `PassivationPersistenceManager`?

I just meant that in normal runtime operation it shouldn't occur due to the checks in the Factory, but like I say I didn't think about how it could be used in tests :slightly_smiling_face:

Reply:
Okay, I can revert it.